tests/core/test_core_edm_copy.cpp

```diff
@@ -0,0 +1,279 @@
+    m_copy.setup(buffer);
+
+    m_copy(vecmem::get_data(host), buffer, vecmem::copy::type::host_to_host);
+
+    compare(vecmem::get_data(host), vecmem::get_data(buffer));
+}
+
+TEST_F(core_edm_copy_test, host_to_device_to_fixed_device_simple) {
+
+    vecmem::edm::host<simple_schema> host{m_resource};
+    fill(host);
+
+    vecmem::edm::buffer<simple_schema> buffer1{
+        static_cast<vecmem::edm::buffer<jagged_schema>::size_type>(host.size()),
+        m_resource, vecmem::data::buffer_type::fixed_size};
+    m_copy.setup(buffer1);
+
+    vecmem::edm::buffer<simple_schema> buffer2{
+        static_cast<vecmem::edm::buffer<jagged_schema>::size_type>(host.size()),
+        m_resource, vecmem::data::buffer_type::fixed_size};
+    m_copy.setup(buffer2);
+
+    m_copy(vecmem::get_data(host), buffer1, vecmem::copy::type::host_to_host);
+    m_copy(buffer1, buffer2, vecmem::copy::type::host_to_host);
+
+    compare(vecmem::get_data(host), vecmem::get_data(buffer2));
+}
+
+TEST_F(core_edm_copy_test, host_to_device_to_fixed_device_jagged) {
+
+    vecmem::edm::host<jagged_schema> host{m_resource};
+    fill(host);
+
+    vecmem::edm::buffer<jagged_schema> buffer1{
+        std::vector<std::size_t>{2, 0, 3}, m_resource, nullptr,
+        vecmem::data::buffer_type::fixed_size};
+    m_copy.setup(buffer1);
+
+    vecmem::edm::buffer<jagged_schema> buffer2{
+        std::vector<std::size_t>{2, 0, 3}, m_resource, nullptr,
+        vecmem::data::buffer_type::fixed_size};
+    m_copy.setup(buffer2);
+
+    m_copy(vecmem::get_data(host), buffer1, vecmem::copy::type::host_to_host);
+    m_copy(buffer1, buffer2, vecmem::copy::type::host_to_host);
+
+    compare(vecmem::get_data(host), vecmem::get_data(buffer2));
+}
+
+TEST_F(core_edm_copy_test, host_to_device_to_resizable_device_simple) {
+
+    vecmem::edm::host<simple_schema> host{m_resource};
+    fill(host);
+
+    vecmem::edm::buffer<simple_schema> buffer1{
+        static_cast<vecmem::edm::buffer<jagged_schema>::size_type>(host.size()),
+        m_resource, vecmem::data::buffer_type::fixed_size};
+    m_copy.setup(buffer1);
+
+    vecmem::edm::buffer<simple_schema> buffer2{
+        10u, m_resource, vecmem::data::buffer_type::resizable};
+    m_copy.setup(buffer2);
+
+    m_copy(vecmem::get_data(host), buffer1, vecmem::copy::type::host_to_host);
+    m_copy(buffer1, buffer2, vecmem::copy::type::host_to_host);
+
+    compare(vecmem::get_data(host), vecmem::get_data(buffer2));
+}
+
+TEST_F(core_edm_copy_test, host_to_device_to_resizable_device_jagged) {
+
+    vecmem::edm::host<jagged_schema> host{m_resource};
+    fill(host);
+
+    vecmem::edm::buffer<jagged_schema> buffer1{
+        std::vector<std::size_t>{2, 0, 3}, m_resource, nullptr,
+        vecmem::data::buffer_type::fixed_size};
+    m_copy.setup(buffer1);
+
+    vecmem::edm::buffer<jagged_schema> buffer2{
+        std::vector<std::size_t>{5, 5, 5}, m_resource, nullptr,
+        vecmem::data::buffer_type::resizable};
```

Comment:
Do you test the resizing somewhere?

Reply:
Not yet. The PR coming after this one, which will be the last one in this saga, will add such tests. (I will need the `vecmem::edm::container` type to make the testing code for that functionality a little less tedious.)